fieldsolver/gridGlue.cpp

```diff
@@ -425,23 +352,23 @@ void getFieldsFromFsGrid(
          ii++;
       }
   }
-  
+
   //post sends
-  sendRequests.resize(onFsgridMapRemoteProcess.size());
+  sendRequests.resize(BgBGrid.coupling->onFsgridMapRemoteProcess.size());
   ii=0;
-  for(auto const &sends: onFsgridMapRemoteProcess){
+  for(auto const &sends: BgBGrid.coupling->onFsgridMapRemoteProcess){
     int remoteRank = sends.first;
     int count = sends.second.size();
     MPI_Isend(sendData[remoteRank].data(), count * sizeof(Average),
 	     MPI_BYTE, remoteRank, 1, MPI_COMM_WORLD,&(sendRequests[ii++]));
   }
-  
+
   MPI_Waitall(receiveRequests.size(), receiveRequests.data(), MPI_STATUSES_IGNORE);
 
 
   //Aggregate receives, compute the weighted average of these
   ii=0;
-  for (auto const &rcv : onDccrgMapRemoteProcess){
+  for (auto const &rcv : BgBGrid.coupling->onDccrgMapRemoteProcess){
```

Comment:
this (and all above references to `BgBGrid.coupling` or `momentsGrid.coupling`) should instead access `technicalGrid.coupling`, I wager